README.md

```diff
@@ -69,29 +67,29 @@ Only the `source` import source phase is specified by this proposal.
 
 ### Defining Module Source
 
-Module source is defined through the [ECMA-262 ES modules HostLoadImportedModule refactoring][],
-which moves the construction of module records to the host, and on which we can define a
-`[[ModuleSourceObject]]` custom object to be returned by module source phase that is compatible
-with module expressions.
+The object provided by the module source phase should have a prototype set to the
+`%AbtractModuleSourcePrototype%` object, defined by this specification to be a minimal
```

Comment:
`Abstract` may be an unnecessary qualifier in this name, since this is to be or become the value of `ModuleSource.prototype` (or `ModuleSource.prototype.__proto__`? Is the intention to allow `ModuleSource.prototype` to be the seat of JavaScript specific features and `ModuleSource.prototype.__proto__ === %AbstractModuleSourcePrototype%`?).

I know we can brand-check module sources by `Object.getOwnPropertyDescriptor(ModuleSource.prototype, Symbol.toStringTag).get.call(subject) !== undefined`, but can one check whether something is an `instanceof` an abstract module source? I think that’s not possible without an abstract module source constructor, like `subject instanceof AbstractModuleSource`, or rather `subject instanceof ModuleSource.prototype.__proto__.constructor`.

Reply:
@kriskowal yes and yes. So `ModuleSource.prototype.__proto__ === %AbstractModuleSourcePrototype%` and `Object.getOwnPropertyDescriptor(ModuleSource.prototype.__proto__, Symbol.toStringTag).get.call(subject) !== undefined` and there's no `AbstractModuleSource` constructor to check.

I'm open to making `AbstractModuleSource` more explicit with a constructor.

Reply:
I don’t think we need a `globalThis.AbstractModuleSource === %AbstractModuleSourceConstructor%` and I think it’s sufficient for:
- `globalThis.ModuleSource.prototype.__proto__ === %AbstractModuleSourcePrototype%`,
- `%AbstractModuleSourcePrototype%.constructor === %AbstractModuleSourceConstructor%`, and
- `%AbstractModuleSourceConstructor%.prototype === %AbstractModuleSourcePrototype%`,
such that `subject instanceof ModuleSource.prototype.__proto__.constructor` can discriminate module source instances with only slightly less fidelity than a brand check.

Reply:
Sure, I can add an AbstractModuleSource constructor for this kind of check at least.

Reply:
I've added an AbstractModuleSource constructor in the latest commit.